### DISCUSSION TITLE
Question regarding a workflow with multiple outputs

### DISCUSSION THREAD

**Discussion by ekageyama** [nextflow-io/nextflow#4643] [Q&A] [answered]
I have a workflow where its last step has a process with multiple channels,  but when I try to use the output, i get the error:
```
ERROR ~ Cannot emit a multi-channel output: $out0
```
the workflow is
```
workflow test{
    take: sample_qch
    main:
        process1(sample_qch)
    emit:
      process1.out
}
```
where the process1 has an output like this
```
        output:
        tuple val(acc), path('filter*')
        path('*.html')
        path('*.json')
```
the idea behind is that the tuple part will keep being processed by a second workflow, while the json and html file will be used for second and third different workflows. Any help is appreciated

**Comment by bentsherman** [comment, marked as answer]
Because your process has multiple output channels, `process1.out` is a multi-channel and cannot be emitted as a workflow output. Instead, you should emit each process output separately in the workflow:
```
emit:
  process1.out[0]
  process1.out[1]
  process1.out[2]
```

And of course you can name those outputs and access them by name, and you can select which outputs you want to emit from the workflow.

**Comment by colindaven** [comment]
I try to avoid tuples for simplicity, but just use name prefixes.

In any event, I write output like this from process1

 ```
   output:
    path "*anno.gff3", emit: anno_gff3
    path "*.sum", emit: summary
```

and if I want to use that output from `process1` in another `process2` then I use this in the `workflow` section:

`process2(process1.out.anno_gff3, process1.out.summary)`
